DataManagementSystem/DB/FileCatalogComponents/FileManager/FileManagerBase.py

```diff
@@ -1068,15 +1068,15 @@ def getReplicaStatus(self, lfns, connection=False):
       return res
     failed = res['Value']['Failed']
     fileIDLFNs = {}
-    for lfn, fileDict in res['Value']['Successful'].iteritems():
+    for lfn, fileDict in res['Value']['Successful'].iteriteritems():
```

Comment:
```suggestion
    for lfn, fileDict in res['Value']['Successful'].iteritems():
```

Reply:
I do not need iteritems(), I am happy with items(). But there is a wide spread opinion that iteritems() is more efficient than items() and we should standardize on that. In this particular case it does not really matter but I would like to understand what should be our preferred style keeping in mind also eventual Python 3 migration.

Reply:
In Python 2 `iteritems` can be notably faster as it avoids copying the data (which is one of the reasons it was renamed `items` in Python 3) but it prevents a single code base from being compatible with both versions at the same. I think the preferred style should be `dict.items()` or, if the dictionary is very large, `six.iteritems(dict)`. The performance will improve in Python 3 regardless of which is used.

### Crude benchmarks for reference

#### Python 2:
```python
In [1]: d = {str(i)*100: {'hello': 'world'} for i in range(1000)}
In [2]: %timeit {k: v for k, v in d.items()}
1000 loops, best of 3: 224 µs per loop
In [3]: %timeit {k: v for k, v in d.iteritems()}
10000 loops, best of 3: 190 µs per loop
````
#### Python 3:
```python
In [1]: d = {str(i)*100: {'hello': 'world'} for i in range(1000)}
In [2]: %timeit {k: v for k, v in d.items()}  # Equivalent to Python 2's iteritems()
72.9 µs ± 1.02 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
In [3]: %timeit {k: v for k, v in list(d.items())}  # Equivalent to Python 2's items()
91.9 µs ± 552 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

For bigger dictionaries the ranking is the same but the relative difference tends to get bigger. With 100000 entries the above benchmarks take 59.8 ms, 42.2 ms, 12.6 ms and 25.7 ms.

Reply:
I tend to disagree with you on the strategy. We started using `iteritem` not only because it is faster, but also to force us to explicitly point out that an iterator is ok (as opposed to a copy), and based on the fact that tools like `2to3` that we intended to use can rely on you making a difference between `item` and `iteritem`. So I don't think that leaving `item` just because it is not much faster is not a good idea. Either we stick with `iteritems` and then rely on future `2to3` passes on the code, or we start using `six.iteritems` everywhere.

Reply:
`2to3` will convert `iteritems` to `six.iteritems`. I think we should change `items()` to `iteritems()` whenever is possible, today, and as @chaen says, not because of the performances.